hs_core/models.py

```diff
@@ -4032,37 +4032,37 @@ def get_crossref_deposit_xml(self, pretty_print=True):
+            """Return funder_uri for a given funder_name from ror funders registry.
+            ROR API Documentation: https://ror.readme.io/docs/api-query
             """
-
-            # url encode the funder name for the query parameter
-            words = funder_name.split()
-            # filter out words that contain the char '.'
-            words = [word for word in words if '.' not in word]
-            encoded_words = [urllib.parse.quote(word) for word in words]
-            # match all words in the funder name
-            query = "+".join(encoded_words)
-            # if we can't find a match in first 50 search records then we are not going to find a match
-            max_record_count = 50
-            email = settings.DEFAULT_DEVELOPER_EMAIL
-            url = f"https://api.crossref.org/funders?query={query}&rows={max_record_count}&mailto={email}"
+            # # url encode the funder name for the query parameter
+            # words = funder_name.split()
+            # # filter out words that contain the char '.'
+            # words = [word for word in words if '.' not in word]
+            # encoded_words = [urllib.parse.quote(word) for word in words]
+            # # match all words in the funder name
+            # query = "+".join(encoded_words)
+            url = "https://api.ror.org/v2/organizations?filter=types:funder"
             funder_name = funder_name.lower()
             response = requests.get(url, verify=False)
             if response.status_code == 200:
                 response_json = response.json()
-                if response_json['status'] == 'ok':
-                    items = response_json['message']['items']
-                    for item in items:
-                        if item['name'].lower() == funder_name:
-                            return item['uri']
-                        for alt_name in item['alt-names']:
-                            if alt_name.lower() == funder_name:
-                                return item['uri']
-                    return ''
+                items = response_json['items']
+                for item in items:
+                    for name in item['names']:
+                        if 'ror_display' in name['types'] and name['value'].lower() == funder_name:
+                            doi_prefix = "https://doi.org/10.13039/"
+                            for external_id in item['external_ids']:
+                                if external_id['type'] == "fundref":
+                                    if external_id['preferred']:
+                                        return doi_prefix + external_id['preferred']
+                                    else:
+                                        all_ids = external_id['all']
+                                        if all_ids:
+                                            return doi_prefix + all_ids[0]
                 return ''
```

Comment:
previously, we returned the "URI" from Crossref, which is of the form: "doi.org/10.13039.XXXXXX"

The way that you're iterating through ROR records will not successfully return the ROR id in many cases. We can use the NSF entry as an example:

Previously, we expected that the `get_funder_id` function would return https://doi.org/10.13039/100000001.
Now that we are using ROR, we expect that it should return https://ror.org/021nxhr62.
However, your implementation of `get_funder_id` is returning https://nsf.gov.
This is because the iterations do not correctly find a matching ID, so `get_funder_id` returns `''` an empty string, and then the `get_crossref_deposit_xml` returns the **agency_url** because the `funder_id` is empty:
https://github.com/hydroshare/hydroshare/blob/2.15.9/hs_core/models.py#L4044